src/transformers/generation/utils.py

```diff
@@ -2258,9 +2192,7 @@ def _contrastive_search(
                 continue  # don't waste resources running the code we don't need
 
             # finished sentences should have their next token be a padding token
-            if eos_token_id is not None:
-                if pad_token_id is None:
-                    raise ValueError("If `eos_token_id` is defined, make sure that `pad_token_id` is defined.")
```

Comment:
In the main `generate` body, we set `pad_token_id` to `eos_token_id` in this situation -- this exception will never be reached